### PR TITLE
Added a lot of clippy suggestions

### PR DESCRIPTION
**Pull request by HotMonkeyWings**
Clippy had suggested a few improvements in the code.

### CODE REVIEW

**Comment by HotMonkeyWings** [comment]
Just 4 warnings remaining, which I don't think we can do much about. (Probably later in the future off the lines of how farzeen suggested to reduce the paramaters)

File: src/peer/mod.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub enum Connection {
     /// Handshake state - handshake with the other peer is in progress
     Handshake,
     /// Connected state - a connection to the other peer has been established
-    Connected(Peer),
+    Connected(Box<Peer>),
```
Comment (aneeshsharma) [inline src/peer/mod.rs:29]: Why change this to `Box<Peer>`?

**Comment by aneeshsharma** [comment]
Also, remove all the error returns inside the handshake thread in `src/peer/mod.rs`
The thread cannot return an error anywhere. Also, there were no return statements earlier so this code would not function properly if any of those errors are encountered.

**Comment by aneeshsharma** [comment]
I think if thats all we can merge this.

> Also, remove all the error returns inside the handshake thread in `src/peer/mod.rs` The thread cannot return an error anywhere. Also, there were no return statements earlier so this code would not function properly if any of those errors are encountered.

I've made this change in #35 as well so no need to do this here.

File: src/peer/handshake.rs [hunk 2]
```diff
@@ -89,28 +86,25 @@ pub fn handshake(
...
-
-                        // Verify the sender has the correct username
-                        if username_recved == peer_username && recved.sequence == recv_seq {
-                            if recved.flags.ack && recved.ack.ack_begin == seq {
-                                break;
-                            }
-                        }
-                    }
+            if let Ok(size) = socket.recv(&mut buf) {
+                let recved = Packet::from(buf[..size].to_vec());
```
[the remark this answers is not in the document]
Comment (HotMonkeyWings) [reply src/peer/handshake.rs]: I guess a simple if inside the if let is the only way, as 'if let' only takes a single condition. For now I made it that

**Pull request merged**